airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/format_command.py

```diff
@@ -0,0 +1,178 @@
+        self.formatter = formatter
+        self.file_filter = file_filter
+        self.get_format_container_fn = get_format_container_fn
+        self.format_commands = format_commands
+        self.export_formatted_code = export_formatted_code
+        self.help = self.get_help_message()
+        self.enable_logging = enable_logging
+        self.exit_on_failure = exit_on_failure
+        self.logger = logging.getLogger(f"Format {self.formatter.value}")
+
+    def get_help_message(self) -> str:
+        """Get the help message for the command.
+
+        Returns:
+            str: The help message
+        """
+        message = f"Run {self.formatter.value} formatter"
+        if self.export_formatted_code:
+            message = f"{message}, will fix any failures."
+        else:
+            message = f"{message}."
+        return message
+
+    def get_dir_to_format(self, dagger_client: dagger.Client) -> dagger.Directory:
+        """Get the directory to format according to the file_filter.
+
+        Args:
+            dagger_client (dagger.Client): The dagger client to use to get the directory
+
+        Returns:
+            dagger.Directory: The directory to format
+        """
+        return dagger_client.host().directory(self.LOCAL_REPO_PATH, include=self.file_filter, exclude=DEFAULT_FORMAT_IGNORE_LIST)
+
+    @pass_pipeline_context
+    @sentry_utils.with_command_context
+    async def invoke(self, ctx: click.Context, click_pipeline_context: ClickPipelineContext) -> Any:
+        """Run the command. If exit_on_failure is True, exit the process with status code 1 if the command fails.
+
+        Args:
+            ctx (click.Context): The click context
+            click_pipeline_context (ClickPipelineContext): The pipeline context
+
+        Returns:
+            Any: The result of running the command
+        """
+        dagger_client = await click_pipeline_context.get_dagger_client(pipeline_name=f"Format {self.formatter.value}")
+        dir_to_format = self.get_dir_to_format(dagger_client)
+        container = self.get_format_container_fn(dagger_client, dir_to_format)
+        command_result = await self.get_format_command_result(dagger_client, container, dir_to_format)
+        if (formatted_code_dir := command_result.output_artifact) and self.export_formatted_code:
+            await formatted_code_dir.export(self.LOCAL_REPO_PATH)
+        if self.enable_logging:
+            log_command_results(ctx, [command_result], main_logger, LogOptions(quiet=ctx.obj["quiet"]))
+        if command_result.status is StepStatus.FAILURE and self.exit_on_failure:
+            sys.exit(1)
+        return command_result
+
+    def disable_logging(self) -> FormatCommand:
+        """Disable logging for the command.
+
+        Returns:
+            FormatCommand: The command with logging disabled
+        """
+        self.enable_logging = False
+        return self
+
+    def dont_exit_on_failure(self) -> FormatCommand:
```

Comment:
This didn't immediately strike me as a setter where its used. As a nit, these could be typical setters like `set_logging_enabled(self, value: bool)` (would rename the prop in that case) and `set_exit_on_failure(self, value: bool)`